app/src/main/java/com/springcamp/rostykboiko/rada3/Main/presenter/MainPresenter.java

```diff
@@ -0,0 +1,139 @@
+package com.springcamp.rostykboiko.rada3.Main.presenter;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.graphics.Rect;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.TypedValue;
+import android.view.View;
+
+import com.springcamp.rostykboiko.rada3.Main.data.MainUseCase;
+import com.springcamp.rostykboiko.rada3.Main.data.MainInteractor;
+import com.springcamp.rostykboiko.rada3.Main.view.MainActivity;
+import com.springcamp.rostykboiko.rada3.MainContract;
+import com.springcamp.rostykboiko.rada3.R;
+import com.springcamp.rostykboiko.rada3.shared.utlils.CardsAdaptor;
+import com.springcamp.rostykboiko.rada3.shared.utlils.OptionListAdapter;
+import com.springcamp.rostykboiko.rada3.shared.utlils.Survey;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MainPresenter implements MainContract.Presenter {
+    @Nullable
+    private CardsAdaptor cardsAdaptor;
+    @Nullable
+    private OptionListAdapter optionsAdapter;
+
+
+    @Nullable
+    private MainContract.View view;
+
+    @Nullable
+    private MainUseCase mainUseCase;
+
+    public MainPresenter(@Nullable MainContract.View view) {
+        this.view = view;
+        this.mainUseCase = new MainInteractor();
+    }
+
+
+    /** List of Cards Start */
+    @Override
+    public RecyclerView cardViewInit(Context mContext, MainContract.View view, CardsAdaptor cardsAdaptor){
+
+        RecyclerView recyclerView = (RecyclerView) view.findViewById(R.id.card_recycler);
+
+        RecyclerView.LayoutManager mLayoutManager = new GridLayoutManager(mContext, 2);
+        recyclerView.setLayoutManager(mLayoutManager);
+        recyclerView.addItemDecoration(new GridSpacingItemDecoration(2, dpToPx(10, mContext), true));
+        recyclerView.setItemAnimator(new DefaultItemAnimator());
+        recyclerView.setAdapter(cardsAdaptor);
+
+        return recyclerView;
+    }
+
+    @Override
+    public CardsAdaptor initCardAdaptor(List<Survey> surveyList, OptionListAdapter optionListAdapter) {
+        cardsAdaptor = new CardsAdaptor(surveyList, optionListAdapter);
+
+        return cardsAdaptor;
+    }
+
+    private class GridSpacingItemDecoration extends RecyclerView.ItemDecoration {
```

Comment:
`GridSpacingItemDecoration` should be inside separate package (for example shared -> view -> GridSpacingItemDecoration), and used insside `Activity/Fragment`